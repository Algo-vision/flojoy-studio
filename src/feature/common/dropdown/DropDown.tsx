import React, { useRef } from "react";
import "@src/feature/common/dropdown/dropdown.css";
interface DropDownProps {
  children: React.ReactNode;
  DropDownBtn: React.ReactNode;
  theme: "light" | "dark";
}
const DropDown = ({ children, DropDownBtn, theme }: DropDownProps) => {
  const DropDownElem = useRef<HTMLDivElement>(null);
  const openDropDown = () => {
    console.log("openDropDown: ",DropDownElem.current);
    if (DropDownElem.current) {
      DropDownElem.current.style.opacity = "1";
      DropDownElem.current.style.zIndex = "50";
      DropDownElem.current.style.transform = "translateY(0)";
    }
  };
  const closeDropDown = () => {
    console.log("closeDropDown: ",DropDownElem.current);
    if (DropDownElem.current) {
      DropDownElem.current.style.opacity = "0";
      DropDownElem.current.style.zIndex = "-1";
      DropDownElem.current.style.transform = "translateY(-10%)";
    }
  };
  return (
    <div
      className="dropdown__wrapper"
      onMouseEnter={openDropDown}
      onMouseLeave={closeDropDown}
    >
      {DropDownBtn}
      <div
        className={`dropdown__container ${theme === "dark" ? "dark" : "light"}`}
        ref={DropDownElem}
        onMouseLeave={closeDropDown}
      >
        {children}
      </div>
    </div>
  );
};

export default DropDown;
