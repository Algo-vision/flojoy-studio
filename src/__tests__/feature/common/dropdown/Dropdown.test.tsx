import "@testing-library/jest-dom/extend-expect"
import "@testing-library/jest-dom"
import { render, fireEvent } from "@testing-library/react";
// import { expect, it } from "@jest/globals";
// import {expect} from 'expect';
import DropDown from "@src/feature/common/dropdown/DropDown";


describe("DropDown", () => {
  const DropDownBtn = <button>Dropdown Button</button>;
  const children = <div>Dropdown Children</div>;
  it("should render correctly", () => {
    const { container, getByText } = render(
      <DropDown DropDownBtn={DropDownBtn} theme="light">
        {children}
      </DropDown>
    );
    // Check that the button is rendered
    const button = getByText("Click me");
    expect(button).toBeInTheDocument();

    // Check that the dropdown content is not visible
    const dropdownContent = getByText("Dropdown content");
    expect(dropdownContent).not.toBeVisible();
    expect(container).toMatchSnapshot();
  });

  //   it("should open and close on hover", () => {
  //     const { getByText, getByTestId } = render(
  //       <DropDown DropDownBtn={DropDownBtn} theme="light">
  //         {children}
  //       </DropDown>
  //     );
  //     const dropdownWrapper = getByTestId("dropdown-wrapper");
  //     const dropdownContainer = getByTestId("dropdown-container");

  //     // Check that dropdown is closed initially
  //     expect(dropdownContainer).toHaveStyle("opacity: 0");
  //     expect(dropdownContainer).toHaveStyle("z-index: -1");
  //     expect(dropdownContainer).toHaveStyle("transform: translateY(-10%)");

  //     // Open dropdown on mouse enter
  //     fireEvent.mouseEnter(dropdownWrapper);
  //     expect(dropdownContainer).toHaveStyle("opacity: 1");
  //     expect(dropdownContainer).toHaveStyle("z-index: 50");
  //     expect(dropdownContainer).toHaveStyle("transform: translateY(0)");

  //     // Close dropdown on mouse leave
  //     fireEvent.mouseLeave(dropdownWrapper);
  //     expect(dropdownContainer).toHaveStyle("opacity: 0");
  //     expect(dropdownContainer).toHaveStyle("z-index: -1");
  //     expect(dropdownContainer).toHaveStyle("transform: translateY(-10%)");
  //   });
});
