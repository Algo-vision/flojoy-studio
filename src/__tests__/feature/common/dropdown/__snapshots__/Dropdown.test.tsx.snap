// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown should render correctly 1`] = `
<div>
  <div
    class="dropdown__wrapper"
  >
    <button>
      Dropdown Button
    </button>
    <div
      class="dropdown__container light"
    >
      <div>
        Dropdown Children
      </div>
    </div>
  </div>
</div>
`;

exports[`DropDown.tsx component should render correctly should match snapshot 1`] = `
<div
  className="dropdown__wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Drop Down
  </button>
  <div
    className="dropdown__container dark"
    onMouseLeave={[Function]}
  >
    <button>
      First Item
    </button>
    <button>
      Second Item
    </button>
  </div>
</div>
`;
